internal/pkg/etcd/certs.go

```diff
@@ -57,3 +76,59 @@ func GeneratePeerCert(etcdCA *x509.PEMEncodedCertificateAndKey) (*x509.PEMEncode
 
 	return x509.NewCertificateAndKeyFromKeyPair(keyPair), nil
 }
+
+// GenerateClientCert generates etcd client certificate and key from etcd CA.
+//
+//nolint:dupl
+func GenerateClientCert(etcdCA *x509.PEMEncodedCertificateAndKey) (*x509.PEMEncodedCertificateAndKey, error) {
+	opts, err := NewCommonOptions()
+	if err != nil {
+		return nil, err
+	}
+
+	opts = append(opts,
+		x509.ExtKeyUsage([]stdlibx509.ExtKeyUsage{
+			stdlibx509.ExtKeyUsageServerAuth,
```

Comment:
should we have both client and server cert to be same cert? or should we split client and server certs? 

not sure what is the best practice for etcd

Reply:
definitely yes. 

But etcd uses this cert to check itself.

* https://kubernetes.io/docs/setup/best-practices/certificates/ uses both (clientAuth, serverAuth)
* https://github.com/kubernetes-sigs/kubespray/blob/master/roles/etcd/templates/openssl.conf.j2 uses both (clientAuth, serverAuth)

I've add talos etcd client cert, now we have 4 certs for etcd cluster.